### ISSUE TITLE
Trigger an optional callback when validation rules change state

### ISSUE THREAD

**Issue by RichTeaTime** [aurelia/validation#214]
The existing ```ValidationGroup.onValidate``` callbacks are only triggered on a manual ```ValidationGroup.validate()``` call, but the setting and unsetting of validity on any added rules is still being performed - based on the binding model - in the background.

This issue suggests adding an extra ```ValidationGroup.onResultPropertyChanged(callback)``` function, which will trigger the callback on each occasion that a ```ValidationResultProperty```'s onValidate callbacks are triggered.

Current implementation is limited by the fact that if any validation rules are added against new binding properties after this call is made, they will not trigger the callback (as it currently just iterates the ```ValidationResultProperty``` objects in ```ValidationResult`` when called).

**Comment by RichTeaTime** [comment]
#208

**Comment by PWKad** [comment]
Thanks for submitting this / commenting on this.  At this time we are closing this because we have completely re-written and are deprecating the previous feature set.  If you feel this should be re-opened please feel free to review [this blog post](http://blog.durandal.io/2016/05/03/aurelias-new-validation-and-testing-capabilities/) and submit either again on this repository, or on the new [validatejs bridge repository](https://github.com/aurelia/validatejs)

Thanks again!
